thirdeye/thirdeye-frontend/app/utils/yaml-tools.js

```diff
@@ -338,12 +339,159 @@ export function postYamlProps(postData) {
+  for (let subscriptionGroup of subscriptionGroups){
+    const detectionConfigIds = subscriptionGroup.detectionConfigIds;
+    for (let id of detectionConfigIds) {
+      let foundAlert = alerts.find(yamlAlert => yamlAlert.id === id);
+      if (foundAlert) {
+        Object.assign(foundAlert, {
+          application: foundAlert.application ? foundAlert.application + ", " + subscriptionGroup.application : subscriptionGroup.application,
+          group: foundAlert.group ? foundAlert.group + ", " + subscriptionGroup.name : subscriptionGroup.name
+        });
+      }
+    }
+  }
+
+  return alerts;
+}
+
+/**
+ * Grab detection type if available, else return yamlAlert.pipelineType
+ */
+function _detectionType(yamlAlert) {
+  if (yamlAlert.rules && Array.isArray(yamlAlert.rules) && yamlAlert.rules.length > 0) {
+    if (yamlAlert.rules[0].detection && Array.isArray(yamlAlert.rules[0].detection) && yamlAlert.rules[0].detection.length > 0) {
+      return yamlAlert.rules[0].detection[0].type;
+    }
+  }
+  return yamlAlert.pipelineType;
+}
+
+// Maps filter name to alert property for filtering
+export const filterToPropertyMap = {
+  application: 'application',
+  subscription: 'group',
+  owner: 'createdBy',
+  type: 'type',
+  metric: 'metric',
+  dataset: 'collection',
+  granularity: 'granularity'
+};
+
+// Maps filter name to alerts API params for filtering
+export const filterToParamsMap = {
+  application: 'application',
+  subscription: 'subscriptionGroup',
+  owner: 'createdBy',
+  type: 'ruleType',
+  metric: 'metric',
+  dataset: 'dataset',
+  granularity: 'granularity',
+  status: 'active',
+  names: 'names'
+};
+
+export function populateFiltersLocal(originalAlerts, rules) {
+  // This filter category is "secondary". To add more, add an entry here and edit the controller's "filterToPropertyMap"
+  const filterBlocksLocal = [
+    {
+      name: 'status',
+      title: 'Status',
+      type: 'checkbox',
+      selected: ['Active', 'Inactive'],
+      filterKeys: ['Active', 'Inactive']
+    },
+    {
+      name: 'application',
+      title: 'Applications',
+      type: 'search',
+      matchWidth: true,
+      hasNullOption: true, // allow searches for 'none'
+      filterKeys: []
+    },
+    {
+      name: 'subscription',
+      title: 'Subscription Groups',
+      hasNullOption: true, // allow searches for 'none'
+      type: 'search',
+      filterKeys: []
+    },
+    {
+      name: 'owner',
+      title: 'Owners',
+      type: 'search',
+      matchWidth: true,
+      filterKeys: []
+    },
+    {
+      name: 'type',
+      title: 'Detection Type',
+      type: 'select',
+      filterKeys: rules
+    },
+    {
+      name: 'metric',
+      title: 'Metrics',
+      type: 'search',
+      filterKeys: []
+    },
+    {
+      name: 'dataset',
+      title: 'Datasets',
+      type: 'search',
+      filterKeys: []
+    },
+    {
+      name: 'granularity',
```

Comment:
seems we can remove this filter in the UI because the endpoint can't perform this search?